managers/backup-manager/BackupManager.js

```diff
@@ -0,0 +1,421 @@
+          operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+          operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+          operationId: result.backup_guid,
+        });
+      })
+      .catch(err => {
+        return Promise
+          .try(() => logger.error(`Error during start of backup - backup to be aborted : ${backupStarted} - backup to be deleted: ${metaUpdated} `, err))
+          .tap(() => {
+            if (registeredStatusPoller) {
+              logger.error(`Error occurred during backup process. Cancelling status poller for deployment : ${deploymentName} and backup_guid: ${backup.guid}`);
+              return ScheduleManager
+                .cancelSchedule(`${deploymentName}_backup_${backup.guid}`,
+                  CONST.JOB.BNR_STATUS_POLLER)
+                .catch((err) => logger.error('Error occurred while performing clean up of backup failure operation : ', err));
+            }
+          })
+          .tap(() => eventmesh.apiServerClient.updateOperationState({
+            resourceId: opts.instance_guid,
+            operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+            operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+            operationId: result.backup_guid,
+            stateValue: CONST.APISERVER.RESOURCE_STATE.ERROR
+          }))
+          .tap(() => eventmesh.apiServerClient.updateOperationResponse({
+            resourceId: opts.instance_guid,
+            operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+            operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+            operationId: result.backup_guid,
+            value: err
+          }))
+          .tap(() => {
+            if (backupStarted) {
+              logger.error(`Error occurred during backup process. Aborting backup on deployment : ${deploymentName}`);
+              return this
+                .abortLastBackup(opts, true)
+                .finally(() => {
+                  if (metaUpdated) {
+                    const options = _
+                      .chain(data)
+                      .pick(data, 'tenant_id', 'backup_guid')
+                      .set('force', true)
+                      .value();
+                    logger.error(`Error occurred during backup process. Deleting backup file on deployment : ${deploymentName} - backup file:`, options);
+                    return this.backupStore
+                      .deleteBackupFile(options);
+                  }
+                })
+                .catch((err) => logger.error('Error occurred while performing clean up of backup failure operation : ', err));
+            }
+          }).then(() => {
+            throw err;
+          });
+      });
+  }
+
+  getOperationState(name, opts) {
+    logger.info(`Retrieving state of last Backup with:`, opts);
+    return this.getBackupOperationState(opts)
+      .then(result => {
+        const deploymentName = opts.deployment;
+        const action = _.capitalize(name);
```

Comment:
why should this be done every time? shouldnt this be capitalized while storing it?